internal/services/automation/client/client.go

```diff
@@ -10,60 +10,31 @@ import (
 	"github.com/hashicorp/go-azure-sdk/resource-manager/automation/2015-10-31/webhook"
 	"github.com/hashicorp/go-azure-sdk/resource-manager/automation/2019-06-01/softwareupdateconfiguration"
 	"github.com/hashicorp/go-azure-sdk/resource-manager/automation/2020-01-13-preview/watcher"
-	"github.com/hashicorp/go-azure-sdk/resource-manager/automation/2022-08-08/dscnodeconfiguration"
-
-	// hybridrunbookworkergroup v2022-08-08 issue: https://github.com/Azure/azure-rest-api-specs/issues/24740
-	runbook2 "github.com/hashicorp/go-azure-sdk/resource-manager/automation/2019-06-01/runbook"
-	"github.com/hashicorp/go-azure-sdk/resource-manager/automation/2021-06-22/hybridrunbookworkergroup"
-	"github.com/hashicorp/go-azure-sdk/resource-manager/automation/2022-08-08/automationaccount"
-	"github.com/hashicorp/go-azure-sdk/resource-manager/automation/2022-08-08/certificate"
-	"github.com/hashicorp/go-azure-sdk/resource-manager/automation/2022-08-08/connection"
-	"github.com/hashicorp/go-azure-sdk/resource-manager/automation/2022-08-08/connectiontype"
-	"github.com/hashicorp/go-azure-sdk/resource-manager/automation/2022-08-08/credential"
-	"github.com/hashicorp/go-azure-sdk/resource-manager/automation/2022-08-08/dscconfiguration"
-	"github.com/hashicorp/go-azure-sdk/resource-manager/automation/2022-08-08/hybridrunbookworker"
-	"github.com/hashicorp/go-azure-sdk/resource-manager/automation/2022-08-08/jobschedule"
-	"github.com/hashicorp/go-azure-sdk/resource-manager/automation/2022-08-08/module"
-	"github.com/hashicorp/go-azure-sdk/resource-manager/automation/2022-08-08/runbook"
-	"github.com/hashicorp/go-azure-sdk/resource-manager/automation/2022-08-08/schedule"
-	"github.com/hashicorp/go-azure-sdk/resource-manager/automation/2022-08-08/sourcecontrol"
-	"github.com/hashicorp/go-azure-sdk/resource-manager/automation/2022-08-08/variable"
+	v20220808 "github.com/hashicorp/go-azure-sdk/resource-manager/automation/2022-08-08"
+	"github.com/hashicorp/go-azure-sdk/sdk/client/resourcemanager"
 	"github.com/hashicorp/terraform-provider-azurerm/internal/common"
 )
 
 type Client struct {
-	AccountClient               *automationaccount.AutomationAccountClient
+	*v20220808.Client
```

Comment:
We should give this a name e.g.
```suggestion
	V20220808 *v20220808.Client
```

Reply:
is this guideline mandatory? because I see `network` is using the embedding too. additionally I believe the embedded form improves readability and facilitates future SDK upgrades.

Reply:
There might need to be some internal discussion on what the convention is there - if we're going with how the meta client is defined in network then we should at least follow the naming convention for the import there which is
`network_2023_04_01.Client`, so in this case `automation_2022_08_08.Client`

Reply:
updated. But I think It's a little bit wordy for `automation` service's sdk with a `automation` prefix. and golang [style guide](https://google.github.io/styleguide/go/decisions.html#package-names) suggests rename the import package name with no underscore.

> Go package names should not have underscores. If you need to import a package that does have one in its name (usually from generated or third party code), it must be renamed at import time to a name that is suitable for use in Go code.

Reply:
I agree with you that since the prefix is already `automation` then there's duplication which could be removed there.

Since this was introduced by @tombuildsstuff there could be a specific reason for doing so which we're not aware of yet (can't check either since he's out at the moment), rather than introduce a different standard into the provider it's better to follow the status quo which will make updating it further down the line quicker since you're only searching for one pattern.

I hope that makes sense.